pymeasure/instruments/agilent/agilentL4534A.py

```diff
@@ -0,0 +1,425 @@
+        value['pre_trig_samples'],
+        value['num_records'],
+        assume_or_convert_units(value['trigger_holdoff'], ureg.s),
+        assume_or_convert_units(value['trigger_delay'], ureg.s)
+    )
+
+
+TRIGGER_OUT_EVENT_VALUES = ['TRIG', 'REC', 'ACQ', 'NONE']
+TRIGGER_OUT_DRIVE_MODE_VALUES = ['POS_50', 'NEG_50', 'POS_25_W_OR', 'OFF']
+
+
+def _validate_trigger_out(value, values):
+    if 'event' not in value:
+        raise ValueError('Requires \'event\'')
+    if 'drive_mode' not in value:
+        raise ValueError('Requires \'drive_mode\'')
+    strict_discrete_set(value['event'], TRIGGER_OUT_EVENT_VALUES)
+    strict_discrete_set(value['drive_mode'], TRIGGER_OUT_DRIVE_MODE_VALUES)
+    return value
+
+
+VOLTAGE_RANGE_VALUES = ureg.Quantity(
+    np.asarray([0.25, 0.5, 1, 2, 4, 8, 16, 32, 128, 256]), ureg.V
+    )
+
+COUPLING_VALUES = ['DC', 'AC']
+FILTER_VALUES = ['LP_20_MHZ', 'LP_2_MHZ', 'LP_200_KHZ']
+
+SAMPLE_RATE_VALUES = ureg.Quantity(
+    np.asarray([
+        1000, 2000, 5000, 10000, 20000, 50000,
+        100000, 200000, 500000, 1000000, 2000000, 5000000, 10000000, 20000000
+    ]), ureg.Hz)
+
+
+def _validate_acq_config(value, values):
+    strict_discrete_set(value['sample_rate'], AgilentL4534A.SAMPLE_RATE_VALUES)
+    strict_discrete_range(value['samples_per_record'], SAMPLE_COUNT_VALUES, 4)
+    strict_discrete_range(value['pre_trig_samples'], PRE_TRIG_VALUES, 4)
+    strict_range(value['num_records'], RECORD_COUNT_VALUES)
+    strict_range(value['trigger_holdoff'], TRIGGER_HOLDOFF_VALUES)
+    strict_range(value['trigger_delay'], TRIGGER_DELAY_VALUES)
+    return value
+
+
+def _validate_channel_config(value, values):
+    if 'range' not in value:
+        raise ValueError('Requires \'range\'')
+    if 'coupling' not in value:
+        raise ValueError('Requires \'coupling\'')
+    if 'filter' not in value:
+        raise ValueError('Requires \'filter\'')
+
+    strict_discrete_set(value['range'], VOLTAGE_RANGE_VALUES)
+    strict_discrete_set(value['coupling'], COUPLING_VALUES)
+    strict_discrete_set(value['filter'], FILTER_VALUES)
+    return value
+
+
+class AgilentL4534A(Instrument):
+    """
+    Represents the Agilent L4532A/L4534A digitizers.
+    """
+    def __init__(self, adapter, name="Agilent L4534A Digitizer", **kwargs):
+        super().__init__(
+            adapter, name, **kwargs
+        )
+
+    SAMPLE_RATE_VALUES = SAMPLE_RATE_VALUES
+
+    class DigitizerChannel(Channel):
+        """Implementation of an Agilent L4532A/L4534A channel."""
+
+        def __init__(self, instrument, id):
+            super().__init__(instrument, id)
+
+        VOLTAGE_RANGE_VALUES = VOLTAGE_RANGE_VALUES
+        COUPLING_VALUES = COUPLING_VALUES
+        FILTER_VALUES = FILTER_VALUES
+
+        config = Instrument.control(
+            "CONF:CHAN:ATTR? (@{ch})",
+            "CONF:CHAN:ATTR (@{ch}),%s",
+            """
+            Control Channel configuration with dict containing range (in V), coupling, and filter.
```

Comment:
Are "range", "coupling", "filter" the keys of the dictionary?

Is it necessary to have this "combined" method or would it not be easier (and more descriptive) to use three properties (at least for setting)?